`LtTestNotVerb` (with `POSTaggerME`) lint is not thread-safe

I'm running this code in many threads
```
new Program(xmir).defects()
```

This is the stack trace:
```
Caused by: java.lang.NullPointerException: Cannot invoke "opennlp.tools.util.Sequence.getOutcomes()" because "this.bestSequence" is null
	at opennlp.tools.postag.POSTaggerME.tag(POSTaggerME.java:150)
	at opennlp.tools.postag.POSTaggerME.tag(POSTaggerME.java:144)
	at org.eolang.lints.misc.LtTestNotVerb.defects(LtTestNotVerb.java:99)
	at org.eolang.lints.misc.LtTestNotVerb.defects(LtTestNotVerb.java:58)
	at org.eolang.lints.Program.defects(Program.java:105)
```

--- a/src/main/java/org/eolang/lints/LtUnlintNonExistingDefect.java
+++ b/src/main/java/org/eolang/lints/LtUnlintNonExistingDefect.java
@@ -120,7 +120,10 @@ final class LtUnlintNonExistingDefect implements Lint {
                     try {
                         return lint.defects(xmir).stream();
                     } catch (final IOException exception) {
-                        throw new IllegalStateException(exception);
+                        throw new IllegalStateException(
+                            String.format("Failed to apply lint %s to XMIR", lint.name()),
+                            exception
+                        );
                     }
                 }
             ).collect(

--- a/src/main/java/org/eolang/lints/MonoLints.java
+++ b/src/main/java/org/eolang/lints/MonoLints.java
@@ -25,7 +25,7 @@ final class MonoLints extends IterableEnvelope<Lint> {
     private static final Iterable<Lint> LINTS = new Shuffled<>(
         new Joined<Lint>(
             new PkByXsl(),
-            MonoLints.mono()
+            MonoLints.javaLints()
         )
     );
 
@@ -62,7 +62,7 @@ final class MonoLints extends IterableEnvelope<Lint> {
      * Java-based lints.
      * @return Java-based lints
      */
-    private static List<Lint> mono() {
+    private static List<Lint> javaLints() {
         try {
             return List.of(
                 new LtAsciiOnly(),
@@ -71,7 +71,10 @@ final class MonoLints extends IterableEnvelope<Lint> {
                 new LtTestNotVerb()
             );
         } catch (final IOException ex) {
-            throw new IllegalArgumentException(ex);
+            throw new IllegalStateException(
+                "Failed to instantiate Java-based lints (ascii-only, reserved-name, test-not-verb)",
+                ex
+            );
         }
     }
 }

--- a/src/main/java/org/eolang/lints/MonoLints.java
+++ b/src/main/java/org/eolang/lints/MonoLints.java
@@ -4,6 +4,7 @@
  */
 package org.eolang.lints;
 
+import java.io.IOException;
 import java.util.List;
 import java.util.stream.Collectors;
 import org.cactoos.iterable.IterableEnvelope;
@@ -24,11 +25,7 @@ final class MonoLints extends IterableEnvelope<Lint> {
     private static final Iterable<Lint> LINTS = new Shuffled<>(
         new Joined<Lint>(
             new PkByXsl(),
-            List.of(
-                new LtAsciiOnly(),
-                new LtReservedName(),
-                new LtSyntaxVersion()
-            )
+            MonoLints.mono()
         )
     );
 
@@ -59,5 +56,22 @@ final class MonoLints extends IterableEnvelope<Lint> {
                 )
             )
         );
+    }
+
+    /**
+     * Java-based lints.
+     * @return Java-based lints
+     */
+    private static List<Lint> mono() {
+        try {
+            return List.of(
+                new LtAsciiOnly(),
+                new LtReservedName(),
+                new LtSyntaxVersion(),
+                new LtTestNotVerb()
+            );
+        } catch (final IOException ex) {
+            throw new IllegalArgumentException(ex);
+        }
     }
 }

--- a/src/main/java/org/eolang/lints/MonoLints.java
+++ b/src/main/java/org/eolang/lints/MonoLints.java
@@ -58,10 +58,6 @@ final class MonoLints extends IterableEnvelope<Lint> {
         );
     }
 
-    /**
-     * Java-based lints.
-     * @return Java-based lints
-     */
     private static List<Lint> javaLints() {
         try {
             return List.of(

--- a/src/main/java/org/eolang/lints/PkByXsl.java
+++ b/src/main/java/org/eolang/lints/PkByXsl.java
@@ -5,6 +5,7 @@
 package org.eolang.lints;
 
 import io.github.secretx33.resourceresolver.PathMatchingResourcePatternResolver;
+import io.github.secretx33.resourceresolver.Resource;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
@@ -75,8 +76,7 @@ final class PkByXsl extends IterableEnvelope<Lint> {
      * @param res Resource with XSL
      * @return Lint
      */
-    private static Lint lint(
-        final io.github.secretx33.resourceresolver.Resource res) {
+    private static Lint lint(final Resource res) {
         try {
             final String url = res.getURL().toString();
             return new LtByXsl(

--- a/src/main/java/org/eolang/lints/PkByXsl.java
+++ b/src/main/java/org/eolang/lints/PkByXsl.java
@@ -48,6 +48,10 @@ final class PkByXsl extends IterableEnvelope<Lint> {
         super(new Shuffled<>(PkByXsl.LINTS));
     }
 
+    /**
+     * Load all lints once.
+     * @return List of all lints
+     */
     @SuppressWarnings("PMD.UnnecessaryLocalRule")
     private static List<Lint> load() {
         try {
@@ -56,29 +60,44 @@ final class PkByXsl extends IterableEnvelope<Lint> {
                     "classpath*:org/eolang/lints/**/*.xsl"
                 )
             ).map(
-                res -> {
-                    try {
-                        final String url = res.getURL().toString();
-                        final String xsl = url.replaceAll(".*org/eolang/lints/", "")
-                            .replaceAll("\\.xsl$", "");
-                        return new LtByXsl(
-                            new InputOf(res.getInputStream()),
-                            new InputOf(
-                                PkByXsl.XSL_PATTERN.matcher(
-                                    PkByXsl.LINTS_PATH.matcher(url).replaceAll("eolang/motives")
-                                ).replaceAll(".md")
-                            ),
-                            new FxResource(
-                                String.format("org/eolang/fixes/%s.xsl", xsl)
-                            )
-                        );
-                    } catch (final IOException ex) {
-                        throw new IllegalArgumentException(ex);
-                    }
-                }
+                PkByXsl::lint
             ).collect(Collectors.toList());
         } catch (final IOException ex) {
-            throw new IllegalArgumentException(ex);
+            throw new IllegalArgumentException(
+                "Failed to load XSL lints from the classpath",
+                ex
+            );
+        }
+    }
+
+    /**
+     * Build a lint for the given XSL resource.
+     * @param res Resource with XSL
+     * @return Lint
+     */
+    private static Lint lint(
+        final io.github.secretx33.resourceresolver.Resource res) {
+        try {
+            final String url = res.getURL().toString();
+            return new LtByXsl(
+                new InputOf(res.getInputStream()),
+                new InputOf(
+                    PkByXsl.XSL_PATTERN.matcher(
+                        PkByXsl.LINTS_PATH.matcher(url).replaceAll("eolang/motives")
+                    ).replaceAll(".md")
+                ),
+                new FxResource(
+                    String.format(
+                        "org/eolang/fixes/%s.xsl",
+                        url.replaceAll(".*org/eolang/lints/", "").replaceAll("\\.xsl$", "")
+                    )
+                )
+            );
+        } catch (final IOException ex) {
+            throw new IllegalArgumentException(
+                "Failed to build a fix for an XSL lint",
+                ex
+            );
         }
     }
 }

--- a/src/main/java/org/eolang/lints/PkByXsl.java
+++ b/src/main/java/org/eolang/lints/PkByXsl.java
@@ -49,11 +49,6 @@ final class PkByXsl extends IterableEnvelope<Lint> {
         super(new Shuffled<>(PkByXsl.LINTS));
     }
 
-    /**
-     * Load all lints once.
-     * @return List of all lints
-     */
-    @SuppressWarnings("PMD.UnnecessaryLocalRule")
     private static List<Lint> load() {
         try {
             return Arrays.stream(
@@ -71,11 +66,6 @@ final class PkByXsl extends IterableEnvelope<Lint> {
         }
     }
 
-    /**
-     * Build a lint for the given XSL resource.
-     * @param res Resource with XSL
-     * @return Lint
-     */
     private static Lint lint(final Resource res) {
         try {
             final String url = res.getURL().toString();

--- a/src/main/java/org/eolang/lints/PkMono.java
+++ b/src/main/java/org/eolang/lints/PkMono.java
@@ -18,11 +18,6 @@ import org.cactoos.list.ListOf;
  * <p>This class is thread-safe.</p>
  *
  * @since 0.23
- * @todo #297:35min Return `LtTestNotVerb` back.
- *  For some reason this lint produces errors in EO-to-Java Compiler. Check
- *  <a href="https://github.com/objectionary/lints/issues/297#issuecomment-2636540673">this</a>
- *  issue for more details. We should return it in the fixed state, once we understand
- *  the root cause of the problem.
  */
 @ThreadSafe
 final class PkMono extends IterableEnvelope<Lint> {

--- a/src/main/java/org/eolang/lints/Vocabulary.java
+++ b/src/main/java/org/eolang/lints/Vocabulary.java
@@ -30,9 +30,14 @@ final class Vocabulary {
     private static final Pattern KEBAB = Pattern.compile("-");
 
     /**
-     * Part-Of-Speech taggers, one per thread.
+     * Part-Of-Speech tagger.
      */
-    private final ThreadLocal<POSTaggerME> taggers;
+    private final POSTaggerME taggers;
+
+    /**
+     * Lock guarding the tagger.
+     */
+    private final java.util.concurrent.locks.ReentrantLock lock;
 
     /**
      * Ctor.
@@ -53,12 +58,8 @@ final class Vocabulary {
      * @param mdl Part-Of-Speech model
      */
     Vocabulary(final POSModel mdl) {
-        this.taggers = new ThreadLocal<>() {
-            @Override
-            protected POSTaggerME initialValue() {
-                return new POSTaggerME(mdl);
-            }
-        };
+        this.taggers = new POSTaggerME(mdl);
+        this.lock = new java.util.concurrent.locks.ReentrantLock();
     }
 
     /**
@@ -71,13 +72,18 @@ final class Vocabulary {
      * @return True if the first word is a VBZ-tagged verb
      */
     boolean isVerb(final String name) {
-        return "VBZ".equals(
-            this.taggers.get().tag(
-                Stream.concat(
-                    Stream.of("It"),
-                    Arrays.stream(Vocabulary.KEBAB.split(name))
-                ).map(s -> s.toLowerCase(Locale.ROOT)).toArray(String[]::new)
-            )[1]
-        );
+        this.lock.lock();
+        try {
+            return "VBZ".equals(
+                this.taggers.tag(
+                    Stream.concat(
+                        Stream.of("It"),
+                        Arrays.stream(Vocabulary.KEBAB.split(name))
+                    ).map(s -> s.toLowerCase(Locale.ROOT)).toArray(String[]::new)
+                )[1]
+            );
+        } finally {
+            this.lock.unlock();
+        }
     }
 }

--- a/src/main/java/org/eolang/lints/Vocabulary.java
+++ b/src/main/java/org/eolang/lints/Vocabulary.java
@@ -7,6 +7,7 @@ package org.eolang.lints;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Locale;
+import java.util.concurrent.locks.ReentrantLock;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
 import opennlp.tools.postag.POSModel;
@@ -59,7 +60,7 @@ final class Vocabulary {
      */
     Vocabulary(final POSModel mdl) {
         this.taggers = new POSTaggerME(mdl);
-        this.lock = new java.util.concurrent.locks.ReentrantLock();
+        this.lock = new ReentrantLock();
     }
 
     /**

--- a/src/main/java/org/eolang/lints/Vocabulary.java
+++ b/src/main/java/org/eolang/lints/Vocabulary.java
@@ -30,9 +30,9 @@ final class Vocabulary {
     private static final Pattern KEBAB = Pattern.compile("-");
 
     /**
-     * Part-Of-Speech tagger.
+     * Part-Of-Speech taggers, one per thread.
      */
-    private final POSTaggerME tagger;
+    private final ThreadLocal<POSTaggerME> taggers;
 
     /**
      * Ctor.
@@ -53,15 +53,12 @@ final class Vocabulary {
      * @param mdl Part-Of-Speech model
      */
     Vocabulary(final POSModel mdl) {
-        this(new POSTaggerME(mdl));
-    }
-
-    /**
-     * Ctor.
-     * @param pos Part-Of-Speech tagger
-     */
-    Vocabulary(final POSTaggerME pos) {
-        this.tagger = pos;
+        this.taggers = new ThreadLocal<>() {
+            @Override
+            protected POSTaggerME initialValue() {
+                return new POSTaggerME(mdl);
+            }
+        };
     }
 
     /**
@@ -75,7 +72,7 @@ final class Vocabulary {
      */
     boolean isVerb(final String name) {
         return "VBZ".equals(
-            this.tagger.tag(
+            this.taggers.get().tag(
                 Stream.concat(
                     Stream.of("It"),
                     Arrays.stream(Vocabulary.KEBAB.split(name))

--- a/src/main/java/org/eolang/lints/Vocabulary.java
+++ b/src/main/java/org/eolang/lints/Vocabulary.java
@@ -33,12 +33,12 @@ final class Vocabulary {
     /**
      * Part-Of-Speech tagger.
      */
-    private final POSTaggerME taggers;
+    private final POSTaggerME tagger;
 
     /**
      * Lock guarding the tagger.
      */
-    private final java.util.concurrent.locks.ReentrantLock lock;
+    private final ReentrantLock lock;
 
     /**
      * Ctor.
@@ -59,7 +59,15 @@ final class Vocabulary {
      * @param mdl Part-Of-Speech model
      */
     Vocabulary(final POSModel mdl) {
-        this.taggers = new POSTaggerME(mdl);
+        this(new POSTaggerME(mdl));
+    }
+
+    /**
+     * Ctor.
+     * @param pos Part-Of-Speech tagger
+     */
+    Vocabulary(final POSTaggerME pos) {
+        this.tagger = pos;
         this.lock = new ReentrantLock();
     }
 
@@ -76,7 +84,7 @@ final class Vocabulary {
         this.lock.lock();
         try {
             return "VBZ".equals(
-                this.taggers.tag(
+                this.tagger.tag(
                     Stream.concat(
                         Stream.of("It"),
                         Arrays.stream(Vocabulary.KEBAB.split(name))

--- a/src/test/java/org/eolang/lints/VocabularyTest.java
+++ b/src/test/java/org/eolang/lints/VocabularyTest.java
@@ -4,9 +4,11 @@
  */
 package org.eolang.lints;
 
+import com.yegor256.Together;
 import java.io.IOException;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.parallel.Execution;
 import org.junit.jupiter.api.parallel.ExecutionMode;
 import org.junit.jupiter.params.ParameterizedTest;
@@ -106,6 +108,34 @@ final class VocabularyTest {
             "Name should be recognized as a verb in singular, but it was not",
             new Vocabulary().isVerb(name),
             Matchers.is(true)
+        );
+    }
+
+    @Test
+    void worksFromMultipleThreads() throws Exception {
+        final Vocabulary vocab = new Vocabulary();
+        MatcherAssert.assertThat(
+            "The shared Vocabulary must give the same answer in all threads",
+            new org.cactoos.set.SetOf<>(
+                new Together<>(
+                    t -> vocab.isVerb("works-as-expected")
+                )
+            ).size(),
+            Matchers.equalTo(1)
+        );
+    }
+
+    @Test
+    void givesConsistentAnswersAcrossThreads() throws Exception {
+        final Vocabulary vocab = new Vocabulary();
+        MatcherAssert.assertThat(
+            "The shared Vocabulary must stay stable under load",
+            new org.cactoos.set.SetOf<>(
+                new Together<>(
+                    t -> vocab.isVerb("works-as-expected")
+                )
+            ).size(),
+            Matchers.equalTo(1)
         );
     }
 }

--- a/src/test/java/org/eolang/lints/VocabularyTest.java
+++ b/src/test/java/org/eolang/lints/VocabularyTest.java
@@ -6,6 +6,7 @@ package org.eolang.lints;
 
 import com.yegor256.Together;
 import java.io.IOException;
+import org.cactoos.set.SetOf;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.jupiter.api.Test;
@@ -116,7 +117,7 @@ final class VocabularyTest {
         final Vocabulary vocab = new Vocabulary();
         MatcherAssert.assertThat(
             "The shared Vocabulary must give the same answer in all threads",
-            new org.cactoos.set.SetOf<>(
+            new SetOf<>(
                 new Together<>(
                     t -> vocab.isVerb("works-as-expected")
                 )
@@ -130,7 +131,7 @@ final class VocabularyTest {
         final Vocabulary vocab = new Vocabulary();
         MatcherAssert.assertThat(
             "The shared Vocabulary must stay stable under load",
-            new org.cactoos.set.SetOf<>(
+            new SetOf<>(
                 new Together<>(
                     t -> vocab.isVerb("works-as-expected")
                 )

